flask_api.py

```diff
@@ -117,14 +128,46 @@ def save_a_recording():
     return filename
 
 
-@app.route('/new_data/courses', methods=['POST'])
+@app.route("/new_data/phrase", methods=["POST"])
+def save_query_phrase():
+    validator = PhrasesValidator()
+    data = request.get_json()
+    try:
+        issues = validator.validate(data)
+    except:
+        return (
+            "Please format the query data: {question: {text: string, variables: list}, answer: {text: string, variables: list}}",
+            BAD_REQUEST,
+        )
+    if issues:
+        try:
+            data = validator.fix(data, issues)
+        except PhrasesValidatorError as err:
+            print("error", err)
+            return str(err), BAD_REQUEST
+
+    db = NimbusMySQLAlchemy(config_file=CONFIG_FILE_PATH)
+    try:
+        phrase_saved = db.insert_entity(QuestionAnswerPair, data)
+    except (BadDictionaryKeyError, BadDictionaryValueError, NimbusDatabaseError) as e:
+        return str(e), BAD_REQUEST
+    except Exception as e:
+        raise e
+
+    if phrase_saved:
+        return "Phrase has been saved", SUCCESS
+    else:
+        return "An error was encountered while saving to database", BAD_REQUEST
```

Comment:
Same comment as above - this is not necessarily a bad request, it could be some other error.  We should probably use a different HTTP error code since something like a database error is not a client error.  See https://developer.mozilla.org/en-US/docs/Web/HTTP/Status for more on HTTP codes (small plug for Mozilla, the makers of the superior web browser).